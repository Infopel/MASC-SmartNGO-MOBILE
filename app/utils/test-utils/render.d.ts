export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
