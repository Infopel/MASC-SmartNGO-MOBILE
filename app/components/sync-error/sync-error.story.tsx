import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { SyncError } from "./sync-error"

storiesOf("SyncError", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <SyncError m='2' />
      </UseCase>
    </Story>
  ))
