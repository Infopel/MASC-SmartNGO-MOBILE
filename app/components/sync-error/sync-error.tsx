import { Text } from "components/text/text"
import { Box, IBoxProps } from "native-base"
import * as React from "react"

export interface SyncErrorProps extends IBoxProps {
  message: string
}

/**
 * Describe your component here
 */
export function SyncError({ message, ...rest }: SyncErrorProps) {



  return (
    <Box bg='cyan.100' {...rest} p='2'>
      <Text>{message}</Text>
    </Box>
  )
}
