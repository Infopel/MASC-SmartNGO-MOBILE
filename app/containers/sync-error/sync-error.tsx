import { SyncError } from "components/sync-error/sync-error"
import { isNil } from "lodash"
import * as React from "react"
import { StyleProp, View, ViewStyle } from "react-native"
import { useSyncDatabase } from "services/sync/sync.hook"

export interface SyncErrorProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
}

/**
 * Describe your component here
 */
export function SyncErrorContainer(props: SyncErrorProps) {
  const { error, isSyncing } = useSyncDatabase()
  return (
    isNil(error) ? < NoErrorSyncView /> : <SyncError message={error}/>
  )
}


export function NoErrorSyncView() {
  return (
    <View />
  )
}