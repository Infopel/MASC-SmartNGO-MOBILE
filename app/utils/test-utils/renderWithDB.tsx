jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  