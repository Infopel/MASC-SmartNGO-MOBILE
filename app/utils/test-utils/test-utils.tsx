import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

