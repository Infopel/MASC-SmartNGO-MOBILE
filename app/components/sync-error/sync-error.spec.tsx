import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { SyncError } from "./sync-error"



describe("<SyncError>",()=>{
  it("render", async function () {
    const component = <SyncError></SyncError>
  
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
