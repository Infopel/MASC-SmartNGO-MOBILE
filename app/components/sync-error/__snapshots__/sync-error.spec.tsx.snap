// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SyncError> render 1`] = `
<View
  dataSet={{}}
  style={
    {
      "backgroundColor": "#cffafe",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 16,
    }
  }
>
  <Text
    dataSet={{}}
    style={
      {
        "backgroundColor": undefined,
        "color": "#171717",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 21,
        "textDecorationLine": undefined,
      }
    }
  >
    Nao foi poss√≠vel sincronizar os dados
  </Text>
</View>
`;
